src/EPR.Calculator.Frontend/Controllers/LocalAuthorityUploadFileProcessingController.cs

```diff
@@ -31,7 +31,7 @@ public IActionResult Index([FromBody] List<LapcapDataTemplateValueDto> lapcapDat
                     throw new ArgumentNullException(lapcapSettingsApi, "LapcapSettingsApi is null. Check the configuration settings for local authority");
                 }
 
-                this.FileName = this.TempData.Peek("LapcapFileName").ToString();
+                this.FileName = this.HttpContext.Session.GetString("LapcapFileName");
```

Comment:
Please create constants for session variables
Also, you need to clean up all the values in try/catch block. Thanks